data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/DynamoDBService.java

```diff
@@ -0,0 +1,289 @@
+import software.amazon.awssdk.services.dynamodb.streams.DynamoDbStreamsClient;
+import software.amazon.awssdk.services.s3.S3Client;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+public class DynamoDBService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DynamoDBService.class);
+
+    private final List<TableConfig> tableConfigs;
+
+    private final Coordinator coordinator;
+
+    private final DynamoDbClient dynamoDbClient;
+    //
+    private final DynamoDbStreamsClient dynamoDbStreamsClient;
+
+    private final S3Client s3Client;
+
+    private final ShardManager shardManager;
+
+    private final ExecutorService executor;
+
+    private final PluginMetrics pluginMetrics;
+
+
+    public DynamoDBService(Coordinator coordinator, ClientFactory clientFactory, DynamoDBSourceConfig sourceConfig, PluginMetrics pluginMetrics) {
+        this.coordinator = coordinator;
+        this.pluginMetrics = pluginMetrics;
+
+        // Initialize AWS clients
+        dynamoDbClient = clientFactory.buildDynamoDBClient();
+        dynamoDbStreamsClient = clientFactory.buildDynamoDbStreamClient();
+        s3Client = clientFactory.buildS3Client();
+
+        // A shard manager is responsible to retrieve the shard information from streams.
+        shardManager = new ShardManager(dynamoDbStreamsClient);
+        tableConfigs = sourceConfig.getTableConfigs();
+        executor = Executors.newFixedThreadPool(3);
+    }
+
+    /**
+     * This service start three long-running threads (scheduler)
+     * Each thread is responsible for one type of job.
+     * The data will be guaranteed to be sent to {@link Buffer} in order.
+     *
+     * @param buffer Data Prepper Buffer
+     */
+    public void start(Buffer<Record<Event>> buffer) {
+
+        LOG.info("Start running DynamoDB service");
+        Runnable exportScheduler = new ExportScheduler(coordinator, dynamoDbClient, s3Client, pluginMetrics);
+        Runnable fileLoaderScheduler = new DataFileScheduler(coordinator, s3Client, pluginMetrics, buffer);
+        Runnable streamScheduler = new StreamScheduler(coordinator, dynamoDbStreamsClient, pluginMetrics, shardManager, buffer);
+
+        // May consider start or shutdown the scheduler on demand
+        // Currently, event after the exports are done, the related scheduler will not be shutdown
+        // This is because in the future we may support incremental exports.
+        executor.submit(exportScheduler);
+        executor.submit(fileLoaderScheduler);
+        executor.submit(streamScheduler);
+
+    }
+
+    /**
+     * Perform graceful shutdown.
+     * Each scheduler must implement logic for interruption.
+     */
+    public void shutdown() {
+        LOG.info("Gracefully shutdown DynamoDB schedulers");
+        executor.shutdownNow();
```

Comment:
Nothing against calling `shutdownNow`, but I don't think it's categorized as graceful shutdown since it forces the thread to quit immediately

Reply:
You are right, I think this is confusing, actually, the scheduler will be interrupped immediately (which is OK), and the scheduler will implement logic to shutdown the real jobs (such as stream consumer) gracefully.  I will remove this.